common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const record = { a: 1, b: 2 };
+const recordA = record.a; // ok: Accessing property 'a' directly, 'a' is defined
+const recordB = record.b; // ok: Accessing property 'b' directly, 'b' is defined
+
+interface NestedIndexProps {
+	nested: {
+		[key: string]: string;
+	};
+}
+type StaticType = { a: string; b: string };
+type IndexSignatureType = { [key: string]: string };
+
+const nestedObj: NestedIndexProps = { nested: { a: "hello" } };
+nestedObj.nested.a; // ok: Accessing nested property 'a', 'a' is defined
+nestedObj.nested.a.length; // defect: Accessing length of a nested undefined property
+
+const a = "a";
+const b = "b";
+const c = "c";
+const someObjWithStaticType: StaticType = { a: "hello", b: "goodbye" };
+someObjWithStaticType.a; // ok: Accessing property 'a', 'a' is defined
+someObjWithStaticType.a.length; // ok: Accessing length of string property 'a', 'a' is a string
+someObjWithStaticType["a"]; // ok: Accessing property 'a' using bracket notation
+someObjWithStaticType["a"].length; // ok: Accessing length of string property 'a' using bracket notation
+someObjWithStaticType[a].length; // ok: Accessing length of string property 'a' using bracket notation
+someObjWithStaticType[b].length; // ok: Accessing length of string property 'b' using bracket notation
+someObjWithStaticType[c].length; // defect: Accessing length of undefined property 'c' using bracket notation
+const aExpectingStringFromStaticType: string = someObjWithStaticType.a; // ok: Assigning string property to a strict string variable, 'a' is defined
+const aExpectingStringOrUndefinedFromStaticType: string | undefined = someObjWithStaticType.a; // ok: Assigning string property to a string or undefined variable
+
+const someObjWithPotenciallyUndefinedProperties: IndexSignatureType = someObjWithStaticType;
```

Comment:
Let's rename this. It has potentially missing properties, not really potentially `undefined` properties. There is a difference.
Maybe just call it `indexedRecordOfStrings`.